src/storage/src/table/storage_table.rs

```diff
@@ -787,7 +790,16 @@ impl<S: StateStore, RS: RowSerde, const T: AccessType> StorageTableBase<S, RS, T
             end_key
         );
 
+        let filter_key = if pk_prefix.size() == 0 {
+            None
+        } else {
+            let pk_prefix_serializer = self.pk_serializer.prefix(pk_prefix.size());
```

Comment:
add assertion to ensure pk_prefix == distribution_key ?

Reply:
Through discussion with eric ， we assume that the current query satisfies the condition of `pk_prefix == distribution_key`, but it is not absolute. Therefore, in order not to affect the existing tests, I hope to introduce this `assert` next pr